docs/static/scripts/Set-AvmGitHubLabels.ps1

```diff
@@ -0,0 +1,205 @@
+
+  [Parameter(Mandatory = $false)]
+  [int]$GitHubCliLimit = 999,
+
+  [Parameter(Mandatory = $false)]
+  [string]$LabelsToApplyCsvUri = "https://azure.github.io/Azure-Verified-Modules/governance/avm-standard-github-labels.csv"
+)
+
+# Check if the GitHub CLI is installed
+$GitHubCliInstalled = Get-Command gh -ErrorAction SilentlyContinue
+if ($null -eq $GitHubCliInstalled) {
+  throw "The GitHub CLI is not installed. Please install the GitHub CLI and try again."
+}
+Write-Host "The GitHub CLI is installed..." -ForegroundColor Green
+
+# Check if GitHub CLI is authenticated
+$GitHubCliAuthenticated = gh auth status
+if ($null -eq $GitHubCliAuthenticated) {
+  throw "Not authenticated to GitHub. Please authenticate to GitHub using the GitHub CLI, `gh auth login`, and try again."
+}
+Write-Host "Authenticated to GitHub..." -ForegroundColor Green
+
+# Check if GitHub repository name is valid
+$GitHubRepositoryNameValid = $RepositoryName -match "^[a-zA-Z0-9-]+/[a-zA-Z0-9-]+$"
+if ($false -eq $GitHubRepositoryNameValid) {
+  throw "The GitHub repository name $RepositoryName is not valid. Please check the repository name and try again. The format must be <OrgName>/<RepoName>"
+}
+
+# List GitHub repository provided and check it exists
+$GitHubRepository = gh repo view $RepositoryName
+if ($null -eq $GitHubRepository) {
+  throw "The GitHub repository $RepositoryName does not exist. Please check the repository name and try again."
+}
+Write-Host "The GitHub repository $RepositoryName exists..." -ForegroundColor Green
+
+# PRE - Get the current GitHub repository labels and export to a CSV file in the current directory or where -OutputDirectory specifies if set to a valid directory path and the directory exists or can be created if it does not exist already
+if ($CreateCsvLabelExports -eq $true) {
+  Write-Host "Getting the current GitHub repository (pre) labels for $RepositoryName..." -ForegroundColor Yellow
+  $GitHubRepositoryLabels = gh label list -R $RepositoryName -L $GitHubCliLimit --json name,description,color
+
+  if ($null -ne $GitHubRepositoryLabels -and $CreateCsvLabelExports -eq $true) {
+    $csvFileNamePathPre = "$OutputDirectory\$($RepositoryName.Replace('/', '_'))-Labels-Pre-$(Get-Date -Format FileDateTime).csv"
+    Write-Host "Exporting the current GitHub repository (pre) labels for $RepositoryName to $csvFileNamePathPre" -ForegroundColor Yellow
+    $GitHubRepositoryLabels | ConvertFrom-Json | Export-Csv -Path $csvFileNamePathPre -NoTypeInformation
+  }
+}
+
+# Remove all pre-existing labels if -RemoveExistingLabels is set to $true and user confirms they want to remove all pre-existing labels
+$GitHubRepositoryLabelsJson = $GitHubRepositoryLabels | ConvertFrom-Json
+if ($null -ne $GitHubRepositoryLabels) {
+  if ($RemoveExistingLabels -eq $true) {
+    $RemoveExistingLabelsConfirmation = Read-Host "Are you sure you want to remove all $($GitHubRepositoryLabelsJson.Count) pre-existing labels from $($RepositoryName)? (Y/N)"
```

Comment:
Does this means that we cannot run the script in automation as it requires user input?

Needs a -Force, or something?